src/System.Memory/src/System/MemoryExtensions.cs

```diff
@@ -222,6 +222,44 @@ public static bool StartsWith<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> valu
             return valueLength <= span.Length && SpanHelpers.SequenceEqual(ref span.DangerousGetPinnableReference(), ref value.DangerousGetPinnableReference(), valueLength);
         }
 
+        /// <summary>
+        /// Determines whether the specified sequence appears at the end of the span.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool EndsWith<T>(this Span<T> span, ReadOnlySpan<T> value)
+            where T : IEquatable<T>
+        {
+            int spanLength = span.Length;
+            int valueLength = value.Length;
+            if (typeof(T) == typeof(byte))
+                return valueLength <= spanLength &&
+                SpanHelpers.EndsWith(
+                    ref Unsafe.As<T, byte>(ref span.DangerousGetPinnableReference()),
+                    ref Unsafe.As<T, byte>(ref value.DangerousGetPinnableReference()),
+                    spanLength,
+                    valueLength);
+            return valueLength <= spanLength && SpanHelpers.EndsWith(ref span.DangerousGetPinnableReference(), ref value.DangerousGetPinnableReference(), spanLength, valueLength);
+        }
+
+        /// <summary>
+        /// Determines whether the specified sequence appears at the end of the span.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool EndsWith<T>(this ReadOnlySpan<T> span, ReadOnlySpan<T> value)
+            where T : IEquatable<T>
+        {
+            int spanLength = span.Length;
+            int valueLength = value.Length;
+            if (typeof(T) == typeof(byte))
+                return valueLength <= spanLength &&
+                SpanHelpers.EndsWith(
```

Comment:
Why not just call SequenceEquals like StartsWith ?

Reply:
You mean something like this?
```csharp
int spanLength = span.Length;
valueLength = value.Length;
(typeof(T) == typeof(byte))
    valueLength <= spanLength &&
    SpanHelpers.SequenceEqual(
         ref Unsafe.As<T, byte>(ref span.Slice(spanLength - valueLength).DangerousGetPinnableReference()),
         ref Unsafe.As<T, byte>(ref value.DangerousGetPinnableReference()),
         valueLength);
return valueLength <= spanLength && SpanHelpers.SequenceEqual(ref span.Slice(spanLength - valueLength).DangerousGetPinnableReference(), ref value.DangerousGetPinnableReference(), valueLength);
```

Reply:
Yes, something like this.

You can use `Unsafe.Add` instead of `span.Slice` to make a bit more efficient and since this is unsafe code anyway.